File: src/main/java/org/hoohoot/homelab/manager/notifications/parser/ArrParser.java
package org.hoohoot.homelab.manager.notifications.parser;

import io.vertx.core.json.JsonArray;

import java.util.Optional;

@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
public abstract class ArrParser {
    protected final Optional<JsonArray> tags;

    protected ArrParser(Optional<JsonArray> tags) {
        this.tags = tags;
    }

    public String requester() {
        return tags.flatMap(tags -> tags.stream().map(Object::toString)
                        .filter(tag -> tag.matches("\\d+ - \\w+"))
                        .map(tag -> tag.split(" - ")[1])
                        .findFirst())
                .orElse("unknown");
    }}

File: src/main/kotlin/org/hoohoot/homelab/manager/domain/ArrParser.kt
package org.hoohoot.homelab.manager.domain

import io.vertx.core.json.JsonArray

fun JsonArray?.requester(): String? =
    this?.map { it.toString() }
        ?.firstOrNull { it.matches("\\d+ - \\w+".toRegex()) }
        ?.split(" - ")
        ?.get(1)

fun String.toImdbLink(): String = "https://www.imdb.com/title/$this/"